io.github.tomluchowski.OpenDungeonsPlus.yml

```diff
@@ -0,0 +1,151 @@
+app-id: io.github.tomluchowski.OpenDungeonsPlus
+runtime: org.freedesktop.Platform
+runtime-version: '24.08'
+sdk: org.freedesktop.Sdk
+command: /app/games/opendungeons-plus
+
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --share=network
+  - --device=dri
+  - --env=LD_LIBRARY_PATH=/app/lib64:/app/lib
```

Comment:
/app/lib64 should not exist, you should install to /app/lib always and fix whatever is getting installed there.

/app/lib is already in ld.so path. So drop this.

Reply:
Sure, I do then             - -DCMAKE_BUILD_INSTALL=/app/lib with the section of CEGUI library, however the program cannot find it .....``tom@localhost:~/Opendungeons_github/flathub> flatpak run io.github.tomluchowski.OpenDungeonsPlus
/app/games/opendungeons-plus: error while loading shared libraries: libCEGUIBase-0.so.2: cannot open shared object file: No such file or directory

Reply:
Place libraries in `/app/lib`, not `/app/lib64`.

Reply:
Are you using flatpak builder 1.4.4 or org.flatpak.Builder ? Libs going in `/app/lib64` could be a side effect of fdsdk dropping patches for cmake and meson in 24.08.

You need flatpak-builder 1.4.4 to compile against 24.08 if you use cmake/meson

Reply:
tom@localhost:~> flatpak-builder --version
flatpak-builder-1.4.2

Reply:
Is really something bad to happen if I put one library in /app/lib64 ?

Reply:
also that's not cmake/meson but cmake/ninja ...

Reply:
# set up install sub-directories
if (CMAKE_SIZEOF_VOID_P EQUAL 8 AND EXISTS "${CMAKE_INSTALL_PREFIX}/lib64")
    set( CEGUI_LIB_INSTALL_DIR lib64 )
elseif (CMAKE_SIZEOF_VOID_P EQUAL 4 AND EXISTS "${CMAKE_INSTALL_PREFIX}/lib32")
    set( CEGUI_LIB_INSTALL_DIR lib32 )
else()
    set( CEGUI_LIB_INSTALL_DIR lib )
endif()
I will fix it asap :), but how ? I don't want anything to be broken , moreover this repo is also connected to the snapcraft build system ....

Reply:
> also that's not cmake/meson but cmake/ninja ...

cmake-ninja is not a different buildsystem than cmake.

> if (CMAKE_SIZEOF_VOID_P EQUAL 8 AND EXISTS "${CMAKE_INSTALL_PREFIX}/lib64")
set( CEGUI_LIB_INSTALL_DIR lib64 )
elseif (CMAKE_SIZEOF_VOID_P EQUAL 4 AND EXISTS "${CMAKE_INSTALL_PREFIX}/lib32")
set( CEGUI_LIB_INSTALL_DIR lib32 )
else()

No need to manually handle it, just use CMAKE_INSTALL_LIBDIR